jdbc-driver/src/main/java/com/google/cloud/dataproc/jdbc/HiveUrlUtils.java

```diff
@@ -0,0 +1,158 @@
+     * @throws IOException
+     * @throws InvalidURLException
+     */
+    public static HiveJdbcConnectionOptions parseHiveUrl(String url) throws InvalidURLException {
+        HiveJdbcConnectionOptions.Builder paramBuilder = HiveJdbcConnectionOptions.builder();
+
+        // Stripped out for the purpose of using URI
+        String urlStripped = url.substring(URL_JDBC_PREFIX.length());
+        URI jdbcURI = URI.create(urlStripped);
+
+        // getPath() will return: /<dbName>;<sessionConfs>, substring gets rid of the "/"
+        String dbAndSessionConfs = jdbcURI.getPath().substring(1);
+        // getQuery() will return the info after '?'
+        String hiveConfs = jdbcURI.getQuery();
+        // getFragmeng() will return the info after '#'
+        String hiveVars = jdbcURI.getFragment();
+
+        String[] params = dbAndSessionConfs.split(";");
+        // Incorrect database name provided
+        if (params.length != 0 && params[0].contains("=")) {
+            throw new InvalidURLException(
+                    String.format(
+                            "'%s' Database name must not contain '='. "
+                                    + "If no dbName is specified, format should be: %s",
+                            params[0], "jdbc:dataproc://hive/;clusterName=<clusterName>"));
+        }
+
+        paramBuilder.setDbName(params.length == 0 ? HIVE_DEFAULT_DATABASE : params[0]);
+        // Params that do not need parsing
+        paramBuilder.setHiveConfs(hiveConfs);
+        paramBuilder.setHiveVars(hiveVars);
+
+        Map<String, String> paramsMap = paramsToMap(params);
+
+        // Get otherSessionConfigs
+        Set<String> nonSessionConfs =
+                ImmutableSet.of(
+                        "projectId",
+                        "region",
+                        "clusterPoolLabel",
+                        "clusterName",
+                        "port",
+                        "httpPath",
+                        "transportMode");
+
+        String otherSessionConfs =
+                paramsMap.entrySet().stream()
+                        .filter(entry -> !nonSessionConfs.contains(entry.getKey()))
+                        .map(Object::toString)
+                        .collect(Collectors.joining(";"));
+        if (!otherSessionConfs.isEmpty()) {
+            paramBuilder.setOtherSessionConfs(otherSessionConfs);
+        }
+
+        // Client shouldn't pass in the session variables that are already set
+        if (paramsMap.containsKey("httpPath") || paramsMap.containsKey("transportMode")) {
+            throw new InvalidURLException(
+                    "Invalid variable.\nPlease do not include httpPath or transportMode.");
+        }
+
+        try {
+            int port =
+                    paramsMap.containsKey("port")
+                            ? Integer.valueOf(paramsMap.get("port"))
+                            : HIVE_DEFAULT_PORT;
```

Comment:
Use `getOrDefault` method (you cam change `HIVE_DEFAULT_PORT` type to string to avoid `String.valueOf(HIVE_DEFAULT_PORT)` call):

```java
Integer.valueOf(paramsMap.getOrDefault("port", String.valueOf(HIVE_DEFAULT_PORT)));
```